### PR TITLE
update falcon with jblas

### PR DESCRIPTION
**Pull request by zhenwei-intel**
## Type of Change

feature

## Description

enable jblas for falcon

## Expected Behavior & Potential Risk

the expected behavior that triggered by this PR 

## How has this PR been tested?

how to reproduce the test (including hardware information)

## Dependency Change?

no

### CODE REVIEW

**Comment by zhenwei-intel** [comment]
```
./bin/main_falcon -m falcon_fp32_new.bin -p "Once upon a time, there existed a little girl, who liked to h
ave adventures. She wanted to" -n 10 -t 28 -b 1024 --seed 12                                                                                                      
main: seed  = 12                                                                                                                                                  
model.cpp: loading model from falcon_fp32_new.bin                                                                                                                 
model.cpp: can't use mmap because tensors are not aligned; convert to new format to avoid this                                                                    
init: n_vocab    = 65024                                                                                                                                          
init: n_ctx      = 512                                                                                                                                            
init: n_embd     = 4544                                                                                                                                           
init: n_mult     = 0                                                                                                                                              
init: n_head     = 71
init: n_layer    = 32
init: n_rot      = 0
init: n_ff       = 18176
init: n_parts    = 1
load: ne ctx size = 27531.44 MB
load: mem required  = 34699.44 MB (+ 4096.00 MB per state)
.....................................................................................
model_init_from_file: kv self size  =  284.00 MB

system_info: n_threads = 28 / 112 | AVX = 1 | AVX2 = 1 | AVX512 = 1 | AVX512_VBMI = 1 | AVX512_VNNI = 1 | FMA = 1 | F16C = 1 | BLAS = 0 | SSE3 = 1 | VSX = 0 | 
sampling: repeat_last_n = 64, repeat_penalty = 1.100000, presence_penalty = 0.000000, frequency_penalty = 0.000000, top_k = 40, tfs_z = 1.000000, top_p = 0.950000, typical_p = 1.000000, temp = 0.800000, mirostat = 0, mirostat_lr = 0.100000, mirostat_ent = 5.000000
generate: n_ctx = 512, n_batch = 1024, n_predict = 10, n_keep = 0


Once upon a time, there existed a little girl, who liked to have adventures. She wanted to travel around the world and see all the wonders it
model_print_timings:        load time = 19328.21 ms
model_print_timings:      sample time =    10.29 ms /    10 runs   (    1.03 ms per token)
model_print_timings: prompt eval time =  3950.16 ms /    20 tokens (  197.51 ms per token)
model_print_timings:        eval time =  2372.40 ms /     9 runs   (  263.60 ms per token)
model_print_timings:       total time = 21953.52 ms
========== eval time log of each prediction ==========
prediction   0, time: 3950.16ms
prediction   1, time: 241.52ms
prediction   2, time: 231.73ms
prediction   3, time: 239.85ms
prediction   4, time: 238.90ms
prediction   5, time: 233.72ms
prediction   6, time: 250.56ms
prediction   7, time: 254.02ms
prediction   8, time: 360.76ms
prediction   9, time: 321.36ms
```

**Comment by zhenwei-intel** [comment]
```
 ./bin/main_falcon -m falcon_q4_0.bin -p "Once upon a time, there existed a little girl, who liked to have 
adventures. She wanted to" -n 10 -t 28 -b 1024 --seed 12                                                                                                          
main: seed  = 12                                                                                                                                                  
model.cpp: loading model from falcon_q4_0.bin                                                                                                                     
init: n_vocab    = 65024                                                                                                                                          
init: n_ctx      = 512                                                                                                                                            
init: n_embd     = 4544                                                                                                                                           
init: n_mult     = 0                                                                                                                                              
init: n_head     = 71                                                                                                                                             
init: n_layer    = 32                                                                                                                                             
init: n_rot      = 0                                                                                                                                              
init: n_ff       = 18176
init: n_parts    = 1
load: ne ctx size =    0.05 MB
load: mem required  = 11040.63 MB (+ 4096.00 MB per state)
.
model_init_from_file: kv self size  =  284.00 MB

system_info: n_threads = 28 / 112 | AVX = 1 | AVX2 = 1 | AVX512 = 1 | AVX512_VBMI = 1 | AVX512_VNNI = 1 | FMA = 1 | F16C = 1 | BLAS = 0 | SSE3 = 1 | VSX = 0 | 
sampling: repeat_last_n = 64, repeat_penalty = 1.100000, presence_penalty = 0.000000, frequency_penalty = 0.000000, top_k = 40, tfs_z = 1.000000, top_p = 0.950000
, typical_p = 1.000000, temp = 0.800000, mirostat = 0, mirostat_lr = 0.100000, mirostat_ent = 5.000000
generate: n_ctx = 512, n_batch = 1024, n_predict = 10, n_keep = 0


Once upon a time, there existed a little girl, who liked to have adventures. She wanted to be everywhere at once, so she would take on
model_print_timings:        load time =  1091.95 ms
model_print_timings:      sample time =    10.34 ms /    10 runs   (    1.03 ms per token)
model_print_timings: prompt eval time =   862.45 ms /    20 tokens (   43.12 ms per token)
model_print_timings:        eval time =   759.94 ms /     9 runs   (   84.44 ms per token)
model_print_timings:       total time =  2087.85 ms
========== eval time log of each prediction ==========
prediction   0, time: 862.45ms
prediction   1, time: 88.12ms
prediction   2, time: 88.87ms
prediction   3, time: 145.42ms
prediction   4, time: 75.32ms
prediction   5, time: 72.23ms
prediction   6, time: 72.17ms
prediction   7, time: 72.38ms
prediction   8, time: 72.89ms
prediction   9, time: 72.53ms
```

**Comment by zhenwei-intel** [comment]
```
./bin/main_falcon -m falcon_q4_j.bin -p "Once upon a time, there existed a little girl, who liked to have 
adventures. She wanted to" -n 10 -t 28 -b 1024 --seed 12                                                                                                          
main: seed  = 12                                                                                                                                                  
model.cpp: loading model from falcon_q4_j.bin                                                                                                                     
init: n_vocab    = 65024                                                                                                                                          
init: n_ctx      = 512                                                                                                                                            
init: n_embd     = 4544                                                                                                                                           
init: n_mult     = 0                                                                                                                                              
init: n_head     = 71
init: n_layer    = 32
init: n_rot      = 0
init: n_ff       = 18176
init: n_parts    = 1
load: ne ctx size =    0.05 MB
load: mem required  = 11464.32 MB (+ 4096.00 MB per state)
.
model_init_from_file: kv self size  =  284.00 MB

system_info: n_threads = 28 / 112 | AVX = 1 | AVX2 = 1 | AVX512 = 1 | AVX512_VBMI = 1 | AVX512_VNNI = 1 | FMA = 1 | F16C = 1 | BLAS = 0 | SSE3 = 1 | VSX = 0 | 
sampling: repeat_last_n = 64, repeat_penalty = 1.100000, presence_penalty = 0.000000, frequency_penalty = 0.000000, top_k = 40, tfs_z = 1.000000, top_p = 0.950000, typical_p = 1.000000, temp = 0.800000, mirostat = 0, mirostat_lr = 0.100000, mirostat_ent = 5.000000
generate: n_ctx = 512, n_batch = 1024, n_predict = 10, n_keep = 0


Once upon a time, there existed a little girl, who liked to have adventures. She wanted to go far away to a land of her own.
model_print_timings:        load time =   524.15 ms
model_print_timings:      sample time =    10.74 ms /    10 runs   (    1.07 ms per token)
model_print_timings:        eval time =   928.92 ms /     9 runs   (  103.21 ms per token)
model_print_timings:       total time =  1686.42 ms
========== eval time log of each prediction ==========
prediction   0, time: 254.27ms
prediction   1, time: 101.75ms
prediction   2, time: 99.50ms
prediction   3, time: 109.06ms
prediction   4, time: 101.42ms
prediction   5, time: 103.05ms
prediction   6, time: 102.75ms
prediction   7, time: 105.86ms
prediction   8, time: 102.63ms
prediction   9, time: 102.89ms
```

**Comment by zhentaoyu** [comment]
> ```
>  ./bin/main_falcon -m falcon_q4_0.bin -p "Once upon a time, there existed a little girl, who liked to have 
> adventures. She wanted to" -n 10 -t 28 -b 1024 --seed 12                                                                                                          
> main: seed  = 12                                                                                                                                                  
> model.cpp: loading model from falcon_q4_0.bin                                                                                                                     
> init: n_vocab    = 65024                                                                                                                                          
> init: n_ctx      = 512                                                                                                                                            
> init: n_embd     = 4544                                                                                                                                           
> init: n_mult     = 0                                                                                                                                              
> init: n_head     = 71                                                                                                                                             
> init: n_layer    = 32                                                                                                                                             
> init: n_rot      = 0                                                                                                                                              
> init: n_ff       = 18176
> init: n_parts    = 1
> load: ne ctx size =    0.05 MB
> load: mem required  = 11040.63 MB (+ 4096.00 MB per state)
> .
> model_init_from_file: kv self size  =  284.00 MB
> 
> system_info: n_threads = 28 / 112 | AVX = 1 | AVX2 = 1 | AVX512 = 1 | AVX512_VBMI = 1 | AVX512_VNNI = 1 | FMA = 1 | F16C = 1 | BLAS = 0 | SSE3 = 1 | VSX = 0 | 
> sampling: repeat_last_n = 64, repeat_penalty = 1.100000, presence_penalty = 0.000000, frequency_penalty = 0.000000, top_k = 40, tfs_z = 1.000000, top_p = 0.950000
> , typical_p = 1.000000, temp = 0.800000, mirostat = 0, mirostat_lr = 0.100000, mirostat_ent = 5.000000
> generate: n_ctx = 512, n_batch = 1024, n_predict = 10, n_keep = 0
> 
> 
> Once upon a time, there existed a little girl, who liked to have adventures. She wanted to be everywhere at once, so she would take on
> model_print_timings:        load time =  1091.95 ms
> model_print_timings:      sample time =    10.34 ms /    10 runs   (    1.03 ms per token)
> model_print_timings: prompt eval time =   862.45 ms /    20 tokens (   43.12 ms per token)
> model_print_timings:        eval time =   759.94 ms /     9 runs   (   84.44 ms per token)
> model_print_timings:       total time =  2087.85 ms
> ========== eval time log of each prediction ==========
> prediction   0, time: 862.45ms
> prediction   1, time: 88.12ms
> prediction   2, time: 88.87ms
> prediction   3, time: 145.42ms
> prediction   4, time: 75.32ms
> prediction   5, time: 72.23ms
> prediction   6, time: 72.17ms
> prediction   7, time: 72.38ms
> prediction   8, time: 72.89ms
> prediction   9, time: 72.53ms
> ```
Does the `prediction   0, time: 862.45ms` mean eval time of token?

**Review by zhentaoyu — APPROVED** [review]
LGTM

**Comment by zhenwei-intel** [comment]
performance on spr (56 core):
q4_0: 38.15 ms per token
q4_j: 148.95 ms per token (gelu takes up a lot of time, when changing gelu to silu, 28.63 ms per token)

File: intel_extension_for_transformers/backends/neural_engine/graph/models/falcon/falcon.cpp [hunk 1]
```diff
@@ -0,0 +1,366 @@
...
+
+#include "core/data_types.h"
+#include "core/ne.h"
+#include "core/ne_layers.h"
+#include "models/model_utils/model_config.h"
+#include "models/model_utils/model_utils.h"
+#include "models/model_utils/util.h"
+
+// feed-forward network
+struct ne_tensor* gpt_neox_ff(const model_layer& layer, ne_context* ctx0, ne_tensor* inp) {
```
[the remark this answers is not in the document]
Comment (zhenwei-intel) [reply intel_extension_for_transformers/backends/neural_engine/graph/models/falcon/falcon.cpp:39]: the function is useless, I will remove it. thanks

**Review by a32543254 — APPROVED** [review]
LGTM

**Pull request merged**